CHANGELOG.md

```diff
@@ -7,10 +7,402 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+            -   `removeUserRole`
+            -   `getRolesForUser`
+            -   `getUsersThatHaveRole`
+        -   Added mandatory `tenantId` in the input for the following recipe interface functions. If any of these functions are overridden, they need to be updated accordingly:
+            -   `addRoleToUser`
+            -   `removeUserRole`
+            -   `getRolesForUser`
+            -   `getRolesForUser`
+    -   Similar changes in combination recipes (thirdpartyemailpassword and thirdpartypasswordless) have been made
+
+### Changes
+
+-   Recipe function changes:
+    -   Added optional `tenantIdForPasswordPolicy` param to `EmailPassword.updateEmailOrPassword`, `ThirdPartyEmailPassword.updateEmailOrPassword`
+    -   Added optional param `tenantId` to `Session.revokeAllSessionsForUser`. If tenantId is undefined, sessions are revoked across all tenants
+    -   Added optional param `tenantId` to `Session.getAllSessionHandlesForUser`. If tenantId is undefined, sessions handles across all tenants are returned
+-   Adds optional param `tenantId` to `getUserCount` which returns total count across all tenants if not passed.
+-   Adds protected prop `tId` to the accessToken payload
+-   Adds `includesAny` claim validator to `PrimitiveArrayClaim`
+
 ### Fixes
 
 -   Fixed an issue where certain Dashboard API routes would return a 404 for Hapi
 
+### Migration
+
+1. To call any recipe function that has `tenantId` added to it, pass `'public`'
+
+Before:
+
+```ts
+EmailPassword.signUp("test@example.com", "password");
+```
+
+After:
+
+```ts
+EmailPassword.signUp("public", "test@example.com", "password");
+```
+
+2. Input for provider array change as follows:
+
+Before:
+
+```ts
+let googleProvider = thirdParty.Google({
+    clientID: "...",
+    clientSecret: "...",
+});
+```
+
+After:
+
+```ts
+let googleProvider = {
+    config: {
+        thirdPartyId: "google",
+        clients: [{ clientId: "...", clientSecret: "..." }],
+    },
+};
+```
+
+3. Single instance with multiple clients of each provider instead of multiple instances of them. Also use `clientType` to differentiate them. `clientType` passed from the frontend will be used to determine the right config.
+
+Before:
+
+```ts
+let providers = [
+    thirdParty.Google({
+        clientID: "clientid1",
+        clientSecret: "...",
+    }),
+    thirdParty.Google({
+        clientID: "clientid2",
+        clientSecret: "...",
+    }),
+];
+```
+
+After:
+
+```ts
+let providers = [
+    {
+        config: {
+            thirdPartyId: "google",
+            clients: [
+                { clientType: "web", clientId: "clientid1", clientSecret: "..." },
+                { clientType: "mobile", clientId: "clientid2", clientSecret: "..." },
+            ],
+        },
+    },
+];
+```
+
+4. Change in the implementation of custom providers
+
+-   All config is part of `ProviderInput`
+-   To update `authorisationRedirect.Params` dynamically, `getAuthorisationRedirectURL` must be overridden
```

Comment:
move it to the next point or move it from here entirely.

Reply:
Done